indexer/contacts_processor.cpp

```diff
@@ -0,0 +1,266 @@
+  {
+    if (instagramPage.front() == '@')
+      return instagramPage.substr(1);
+    else
+      return instagramPage;
+  }
+  if (EditableMapObject::ValidateWebsite(instagramPage))
+  {
+    string instagramPageUrl = instagramPage;
+    // Check if HTTP protocol is present.
+    if (!strings::StartsWith(instagramPage, "http://") && !strings::StartsWith(instagramPage, "https://"))
+      instagramPageUrl = "https://" + instagramPage;
+
+    const url::Url url = url::Url(instagramPageUrl);
+    const string & domain = strings::MakeLowerCase(url.GetWebDomain());
+    // Check Instagram domain name.
+    if (domain == "instagram.com" || strings::EndsWith(domain, ".instagram.com"))
+    {
+      auto webPath = url.GetWebPath();
+      // Strip last '/' symbol.
+      if (webPath.back() == '/')
+        return webPath.substr(0, webPath.length()-1);
+      else
+        return webPath;
+    }
+  }
+
+  return {};
+}
+
+string ValidateAndFormat_twitter(string const & twitterPage)
+{
+  if (twitterPage.empty())
+    return {};
+  // Check that twitterPage contains valid username.
+  // Rules took here: https://stackoverflow.com/q/11361044
+  if (regex_match(twitterPage, regex(R"(^@?[A-Za-z0-9_]{1,15}$)")))
+  {
+    if (twitterPage.front() == '@')
+      return twitterPage.substr(1);
+    else
+      return twitterPage;
+  }
+  if (EditableMapObject::ValidateWebsite(twitterPage))
+  {
+    string twitterPageUrl = twitterPage;
+    // Check if HTTP protocol is present.
+    if (!strings::StartsWith(twitterPage, "http://") && !strings::StartsWith(twitterPage, "https://"))
+      twitterPageUrl = "https://" + twitterPage;
+
+    const url::Url url = url::Url(twitterPageUrl);
+    const string & domain = strings::MakeLowerCase(url.GetWebDomain());
+    // Check Twitter domain name.
+    if (domain == "twitter.com" || strings::EndsWith(domain, ".twitter.com"))
+    {
+      auto webPath = url.GetWebPath();
+      // Strip last '/' symbol.
+      if (webPath.back() == '/')
+        webPath = webPath.substr(0, webPath.length()-1);
+
+      // Strip first '@' symbol
+      if (webPath.front() == '@')
+        webPath = webPath.substr(1);
+
+      return webPath;
+    }
+  }
+
+  return {};
+}
+
+string ValidateAndFormat_vk(string const & vkPage)
+{
+  if (vkPage.empty())
+    return {};
+  {
+    /* Check that vkPage contains valid page name. Rules are defined here: https://vk.com/faq18038
+     * The page name must be between 5 and 32 characters.
+       Invalid format could be in cases:
+     * - begins with three or more numbers (one or two numbers are allowed).
+     * - begins and ends with "_".
+     * - contains a period with less than four symbols after it starting with a letter.
+     */
+
+    string vkPageClean = vkPage;
+    if (vkPageClean.front() == '@')
+      vkPageClean = vkPageClean.substr(1);
+
+    if (vkPageClean.front() == '_' && vkPageClean.back() == '_') return {};
```

Comment:
nit: easier to debug if the "if's" block (return here) is on the next line. And it is also in our style guide.

Reply:
Moved return to next line